documentation/docs/05-load.md

```diff
@@ -2,65 +2,118 @@
 
-### Input properties
-
-The argument to a `load` function is a `LoadEvent` (or, for server-only `load` functions, a `ServerLoadEvent` which inherits `clientAddress`, `cookies`, `locals`, `platform` and `request` from `RequestEvent`). All events have the following properties:
+```svelte
+/// file: src/routes/blog/+page.svelte
+<script>
+	/** @type {import('./$types').PageData} */
+	export let data;
+</script>
 
-#### data
+{#each data.posts as post}
+	...
+{/each}
+```
 
-Very rarely, you might need both a `+page.js` and a `+page.server.js` (or the `+layout` equivalent). In these cases, the `data` for `+page.svelte` comes from `+page.js`, which in turn receives `data` from the server:
+You can do the same for layouts:
 
 ```js
-/// file: src/routes/my-route/+page.server.js
-/** @type {import('./$types').PageServerLoad} */
-export function load() {
+/// file: src/routes/blog/+layout.server.js
+// @filename: ambient.d.ts
+declare const db: { getBlogPostsCount: () => Promise<any[]> }
+
+// @filename: index.js
+// ---cut---
+/** @type {import('./$types').LayoutServerLoad} */
+export async function load() {
 	return {
-		a: 1
+		postsCount: await db.getBlogPostsCount()
 	};
 }
 ```
 
-```js
-/// file: src/routes/my-route/+page.js
-// @filename: $types.d.ts
-export type PageLoad = import('@sveltejs/kit').Load<{}, { a: number }>;
+```svelte
+/// file: src/routes/blog/+layout.svelte
+<script>
+	/** @type {import('./$types').LayoutData} */
+	export let data;
+</script>
 
-// @filename: index.js
-// ---cut---
-/** @type {import('./$types').PageLoad} */
-export function load({ data }) {
+<span>{data.postsCount} blog posts available</span>
+<slot />
+```
+
+Top-level promises will be awaited, which makes it easy to return multiple promises without creating a waterfall:
+
+```js
+/// file: src/routes/+page.server.js
+/** @type {import('./$types').PageServerLoad} */
+export function load() {
 	return {
-		b: data.a * 2
+		a: Promise.resolve('a'),
+		b: Promise.resolve('b'),
+		c: {
+			value: Promise.resolve('c')
+		}
 	};
 }
 ```
 
 ```svelte
-/// file: src/routes/my-route/+page.svelte
 <script>
 	/** @type {import('./$types').PageData} */
 	export let data;
 
-	console.log(data.a); // `undefined`, it wasn't passed through in +page.js
-	console.log(data.b); // `2`
+	console.log(data.a); // 'a'
+	console.log(data.b); // 'b'
+	console.log(data.c.value); // `Promise {...}`
 </script>
 ```
 
-In other words `+page.server.js` passes `data` along to `+page.js`, which passes `data` along to `+page.svelte`.
+The `load` functions related to a certain page run in parallel by default, avoiding a waterfall of requests. Once all `load` functions have returned, rendering starts. This allows layouts and pages to access data from upper layouts. In the above example, the `data` prop in `blog/+page.svelte` _also_ has access to the `postsCount` returned from `blog/+layout.server.js`. Additionally, you can access the merged data of _all_ `load` functions through `$page.data`.
```

Comment:
Oh, I didn't realize you could get layout data from the `data` prop directly. Does this include all nested layouts as well for a page?

Reply:
Yes, everything above the layout/page is merged into it. Think if it like `$page.data` which stops at the layout/page

Reply:
I think this could use more clarification in the docs, then, since this was a surprise to me! I thought page.data was the only way to get all merged data.